auth_brute_force/models/res_authentication_attempt.py

```diff
@@ -2,35 +2,160 @@
 # Copyright 2015 GRAP - Sylvain LE GAL
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from odoo import _, api, fields, models
+import json
+import logging
+from urllib2 import urlopen
+from odoo import api, fields, models
+from odoo.osv.expression import TRUE_LEAF
+
+GEOLOCALISATION_URL = u"http://ip-api.com/json/{}"
```

Comment:
Not sure about pros/cons of existing online service alternatives, but maybe it would be better to use a service which has HTTPS in their free plan, eg. https://ipapi.co/json?

Also, I have tested http://ip-api.com/json/ locally and after a couple of hundreds of requests I got banned (I guess timeout based), so in a real world case I believe this could potentially break if the limit is reached within the time window, resulting in empty metadata?

IIUC this would also be stored upon successful logins? In that case, each new `res.authentication.attempt` record could potentially take 5+ seconds? What if we try to fill the metadata asynchronously afterwards, eg. with Cron or connector? This way you could also group records by remote address and reduce the number of requests in this way  (now it is repeated for the same IP).

Also, I am wondering why an external service is used here instead of eg. [GeoIP](https://github.com/odoo/odoo/blob/14f59f4b780fa1e757cfce6a46c62c7a6b00d13e/addons/website/geoipresolver.py#L16)?

Reply:
Hi. I implemented the call of ip-api some years ago, but I did'nt made a benchmark about all the solutions, so maybe it's not the best solution.
Fill the metadata via a cron could be a great improvment indeed. And using GeoIP too ! (but maybe not the objective of this PR).

Reply:
I agree with both: this should be improved, but this PR targets other purpose. I added it to known issues.